backends/etcd/etcd.go

```diff
@@ -163,6 +241,39 @@ Nodes:
 	return sx, nil
 }
 
+func (g *Backendv3) loopNodes(kv []*mvccpb.KeyValue, nameParts []string, star bool, bx map[bareService]bool) (sx []msg.Service, err error) {
+	if bx == nil {
+		bx = make(map[bareService]bool)
+	}
+
+	index := 0
+	lengthOfEntries := len(kv)
+	for index < lengthOfEntries {
+
+		serviceInstance := new(msg.Service)
+
+		if err := json.Unmarshal(kv[index].Value, serviceInstance); err != nil {
+			return nil, err
+		}
+
+		b := bareService{serviceInstance.Host, serviceInstance.Port, serviceInstance.Priority, serviceInstance.Weight, serviceInstance.Text}
+
+		bx[b] = true
+		serviceInstance.Key = string(kv[index].Key)
+		//TODO: shouldn't be that another call (LeaseRequest) for TTL
+		serviceInstance.Ttl = g.calculateTtl(kv[index], serviceInstance)
+
+		if serviceInstance.Priority == 0 {
+			serviceInstance.Priority = int(g.config.Priority)
+		}
+
+		sx = append(sx, *serviceInstance)
```

Comment:
why is this unconditionally added? Why are you not checking the wildcard parts of the name here? I..e what the loopNodes for version 2 does...